catalog/app/containers/Bucket/requests/packageUpload.ts

```diff
@@ -0,0 +1,258 @@
+import * as R from 'ramda'
+import * as React from 'react'
+
+import { JsonValue } from 'components/JsonEditor/constants'
+import * as APIConnector from 'utils/APIConnector'
+import { makeSchemaDefaultsSetter, JsonSchema } from 'utils/json-schema/json-schema'
+import pipeThru from 'utils/pipeThru'
+import * as workflows from 'utils/workflows'
+
+// "CREATE package" - creates package from scratch to new target
+// "UPDATE package" - creates package from source (existing manifest) to the same target
+// "COPY package" - creates package from source (existing manifest) to new target
+// "WRAP package" - creates package (wraps directory) from source (bucket + directory) to new target
+
+interface FileEntry {
+  is_dir: boolean
+  logical_key: string
+  path: string
+}
+
+interface FileUpload {
+  hash: string
+  logical_key: string
+  physical_key: {
+    bucket: string
+    key: string
+    version: string
+  }
+  size: number
+}
+
+interface ManifestBodyBase {
```

Comment:
it's not exactly "Manifest", i'd use a more generic name like "BaseRequestBody", "CreateRequestBody", etc

Reply:
"request body" just means nothing to me, but I'm ok with it since it's inside one module

Reply:
yep, we get the meaning from the context